core/src/main/scala/org/apache/spark/api/java/JavaUtils.scala

```diff
@@ -43,10 +43,16 @@ private[spark] object JavaUtils {
 
     override def size: Int = underlying.size
 
-    override def get(key: AnyRef): B = try {
-      underlying.getOrElse(key.asInstanceOf[A], null.asInstanceOf[B])
-    } catch {
-      case ex: ClassCastException => null.asInstanceOf[B]
+    // Delegate to implementation because AbstractMap implementation iterates over whole key set
+    override def containsKey(key: AnyRef): Boolean = {
+      key.isInstanceOf[A] && underlying.contains(key.asInstanceOf[A])
+    }
+
+    override def get(key: AnyRef): B = {
+      if (key.isInstanceOf[A]) {
+        return underlying.getOrElse(key.asInstanceOf[A], null.asInstanceOf[B])
```

Comment:
It's more conventional to avoid `return`, and just write if () { ... foo } else { ... bar }. No big deal.